### ISSUE TITLE
Preview documentation from Pull Requests

### ISSUE THREAD

**Issue by dgarcia360** [scylladb/sphinx-scylladb-theme#277]
## Motivation

Every time contributors send pull requests that update the docs, maintainers have to download and build the documentation locally to preview the changes.

The docs toolchain could build a preview site for pull requests. For example, when a repository receives a new pull request that edits the contents inside the docs folder, it should create a preview for this pull request.

Once the build completes, the pull request should receive a message with the URL to preview the docs.
## Requirements

- All the pull-requests previews will be publicly available.
- Only build previews for the current version (without multiversion)
- Only build pull requests previews if the doc folder receives an update.
- Remove pull requests preview after the PR is closed to reduce hosting costs.

## Implementation

We have selected Netlify CI as the service to generate the [documentation previews](https://davidgarcia.dev/posts/previewing-documentation-in-pull-request/).

First, we will start with a pilot project adding the selected service to the repository sphinx-scylladb-theme. Then, we can gradually implement this solution in other repositories if the pilot project completes successfully.

### Initial Setup

Prerequisite: Remove Makefile dependency https://github.com/scylladb/sphinx-scylladb-theme/issues/347

1. Add a new command to preinstall CI requirements.
2. Edit GitHub Actions scripts to use the CI setup command.
3. Pilot: Add Netlify CI in one repo (sphinx-scylladb-theme)
4. Test and evaluate the solution
